lib/db/lowlevel.go

```diff
@@ -573,6 +577,21 @@ func (db *Lowlevel) gcIndirect() error {
 	// Indirection GC needs to run when there are no modifications to the
 	// FileInfos or indirected items.
 
+	// checkCtx is a helper that periodically checks the ctx for Done() and
+	// if so returns an error.
```

Comment:
Is the "ctx-check-throttling" actually useful? I'd expect receiving from `ctx.Done()` to be fairly cheap and we do quite a few operations in between. I haven't been shy throwing context checks into database iterations, so if it is relevant that should probably be changed too.

Reply:
I’ll check, hopefully you’re right.

Reply:
It's about 5% of a bloom filter lookup, so not *completely* negligible in a tight loop, but in the middle of a database iteration that might (worst case) involve I/O I think you're right and it is negligible. I'll remove the complexities.